### PR TITLE
ADEN-5610 Added Casper tests for ad params send to Qubole

### PR DESCRIPTION
**Pull request by bulaszenko**
Added Casper tests for ad params send to Qubole

### CODE REVIEW

**Comment by drets** [comment]
So many repos I am watching :see_no_evil: :tada:

**Comment by drets** [comment]
Could we add `node_modules` folder to `.gitignore`? :smile:

File: tests/testConfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "url":
+  {
+    "home": "http://sandbox-adeng05.project43.wikia.com/wiki/Project43_Wikia"
+  },
+  "instantGlobals":
+  {
+    "hivi": "?InstantGlobals.wgAdDriverHiViLeaderboardRabbitCountries=[XX]"
+  }
+}
```
Comment (v3rth) [inline tests/testConfig.json]: Add new line

File: tests/testConfig.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "url":
+  {
+    "home": "http://sandbox-adeng05.project43.wikia.com/wiki/Project43_Wikia"
```
Comment (v3rth) [inline tests/testConfig.json]: I think only "host" part should be inside test config file, rest should be placed inside test case

File: tests/testConfig.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (v3rth) [inline tests/testConfig.json]: Maybe try to rename this file to `testConfig.default.json`, or something like this, and this one add to `.gitignore`

File: tests/testHomeTopAdParamsSendToQubole.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            });
+
+            dataParams.homeAdInfoRegexPaterns.forEach(function (testCase) {
+                var expectedValue = new RegExp(testCase.expectedPattern);
+                test.assertMatch(urlDataJson[testCase.attr], expectedValue, testCase.attr + ": correct");
+            });
+        }
+    });
+
+    casper.start(config.url.home + config.instantGlobals.hivi).run(function () {
```
Comment (v3rth) [inline tests/testHomeTopAdParamsSendToQubole.js]: Let's add here const and don't use variable from config, it shouldn't be modifiable.

File: tests/testHomeTopAdParamsSendToQubole.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+var helper = require("helper.js"),
+    utils = require("utils"),
+    config = require('testConfig.json'),
+    dataParams = require('quboleAdParameters.json'),
+    adInfoRegex = /http:\/\/a\.wikia-beacon\.com\/__track\/special\/adengadinfo\?.*TOP_LEADERBOARD.*/;
+
+casper.test.begin('test oasis home ad params send to Qubole', function (test) {
+
+    casper.on('resource.requested', function(resource) {
```
Comment (v3rth) [inline tests/testHomeTopAdParamsSendToQubole.js]: It still looks like a part of external library, so let's extract it, but as parameters we should be able to add from perspective of test function with assertion

(but I'm not sure if it is possible to do it clearly :disappointed: )

File: tests/testHomeTopAdParamsSendToQubole.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+        if (url !== null) {
+            var adInfoUrl = url.toString(),
+                urlDataJson = helper.getUrlToJson(adInfoUrl);
+
+            dataParams.homeAdInfoStringPaterns.forEach(function (testCase) {
+                test.assertEquals(urlDataJson[testCase.attr], testCase.expectedString, testCase.attr + ": correct");
+            });
+
+            dataParams.homeAdInfoRegexPaterns.forEach(function (testCase) {
```
Comment (v3rth) [inline tests/testHomeTopAdParamsSendToQubole.js]: It looks like something what are you able to extract and unify with previous forEach (It's not "must" have, keep in mind clarity)

File: tests/testHomeTopAdParamsSendToQubole.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+                var expectedValue = new RegExp(testCase.expectedPattern);
+                test.assertMatch(urlDataJson[testCase.attr], expectedValue, testCase.attr + ": correct");
+            });
+        }
+    });
+
+    casper.start(config.url.home + config.instantGlobals.hivi).run(function () {
+        casper.test.done();
+    });
+});
```
Comment (v3rth) [inline tests/testHomeTopAdParamsSendToQubole.js]: New line, please :)

File: tests/testHomeTopAdParamsSendToQubole.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    adInfoRegex = /http:\/\/a\.wikia-beacon\.com\/__track\/special\/adengadinfo\?.*TOP_LEADERBOARD.*/;
+
+casper.test.begin('test oasis home ad params send to Qubole', function (test) {
+
+    casper.on('resource.requested', function(resource) {
+        var url = resource.url.match(adInfoRegex);
+
+        if (url !== null) {
+            var adInfoUrl = url.toString(),
+                urlDataJson = helper.getUrlToJson(adInfoUrl);
```
Comment (v3rth) [inline tests/testHomeTopAdParamsSendToQubole.js]: I think it is easier to just use `resource.url` without [creating `adInfoUrl` variable](https://github.com/Wikia/ad-tests/pull/14/files#diff-619212a0abcd177ec45ca0be7b8e43aaR13)

File: tests/testHomeTopAdParamsSendToQubole.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    config = require('testConfig.json'),
+    dataParams = require('quboleAdParameters.json'),
+    adInfoRegex = /http:\/\/a\.wikia-beacon\.com\/__track\/special\/adengadinfo\?.*TOP_LEADERBOARD.*/;
+
+casper.test.begin('test oasis home ad params send to Qubole', function (test) {
+
+    casper.on('resource.requested', function(resource) {
+        var url = resource.url.match(adInfoRegex);
+
+        if (url !== null) {
```
Comment (v3rth) [inline tests/testHomeTopAdParamsSendToQubole.js]: If you refactor a bit, you will be able to [use match function](https://github.com/Wikia/ad-tests/pull/14/files#diff-619212a0abcd177ec45ca0be7b8e43aaR10) without assigning to `url` variable

File: .travis.yml [hunk 10]
```diff
@@ -1,17 +1,17 @@
...
-install: pip install -r requirements.txt
+install:
+   - npm install
 
 before_script:
-  - flake8 .
+   - sleep 5
 
-script: python ad_tests.py
+script: npm run casper
```
Comment (v3rth) [inline .travis.yml]: new line PLZ

**Pull request merged**